components/lifter.py

```diff
@@ -1,63 +1,99 @@
+    SWITCH_HEIGHT = 0.47625 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
 
     def setup(self):
         """This is called after variables are injected by magicbot."""
-        self.set_pos = self.GROUND_HEIGHT
+        self.set_pos = None
+        self.default_switch = None
         self.default_height = self.UPPER_SCALE
 
-        self.lift_motor.overrideLimitSwitchesEnable(True)
-        #  TODO tune motion profiling
-        self.lift_motor.configMotionAcceleration(100, 10)
-        self.lift_motor.configMotionCruiseVelocity(775, 10)
+        self.motor.setInverted(True)
+        self.motor.setNeutralMode(ctre.WPI_TalonSRX.NeutralMode.Brake)
+
+        self.motor.overrideLimitSwitchesEnable(False)
+        self.motor.configReverseLimitSwitchSource(ctre.WPI_TalonSRX.LimitSwitchSource.FeedbackConnector, ctre.WPI_TalonSRX.LimitSwitchNormal.NormallyOpen, deviceID=0, timeoutMs=10)
+        self.motor.configForwardLimitSwitchSource(ctre.WPI_TalonSRX.LimitSwitchSource.Deactivated, ctre.WPI_TalonSRX.LimitSwitchNormal.Disabled, deviceID=0, timeoutMs=10)
+
+        self.motor.overrideSoftLimitsEnable(True)
+        self.motor.configForwardSoftLimitEnable(True, timeoutMs=10)
+        self.motor.configForwardSoftLimitThreshold(int(self.TOP_HEIGHT), timeoutMs=10)
+        self.motor.configReverseSoftLimitEnable(True, timeoutMs=10)
+        self.motor.configReverseSoftLimitThreshold(self.BOTTOM_HEIGHT, timeoutMs=10)
+
+        self.motor.configSelectedFeedbackSensor(ctre.WPI_TalonSRX.FeedbackDevice.QuadEncoder, 0, timeoutMs=10)
+
+        # TODO tune motion profiling
+        self.motor.selectProfileSlot(0, 0)
+        self.motor.config_kF(0, 1023/self.FREE_SPEED, timeoutMs=10)
+        self.motor.config_kP(0, 0.7, timeoutMs=10)
+        self.motor.config_kI(0, 0.01, timeoutMs=10)
+        self.motor.config_kD(0, 0, timeoutMs=10)
+
+        self.motor.configMotionAcceleration(500, timeoutMs=10)
+        self.motor.configMotionCruiseVelocity(self.FREE_SPEED, timeoutMs=10)
 
     def on_enable(self):
         """This is called whenever the robot transitions to being enabled."""
-        self.reset()
 
     def on_disable(self):
         """This is called whenever the robot transitions to disabled mode."""
         self.stop()
 
     def execute(self):
         """Run at the end of every control loop iteration."""
-        pass
+        if self.set_pos is not None:
+            self.motor.set(self.motor.ControlMode.MotionMagic, self.set_pos)
+
+            if self.at_pos():
+                self.stop()
 
     def stop(self):
         """Stop the lift motor"""
-        self.lift_motor.stopMotor()
+        self.set_pos = None
+        self.motor.stopMotor()
 
     def reset(self):
-        self.set_pos = self.GROUND_HEIGHT * self.DISTANCE_PER_COUNT
-        self.lift_motor.set(mode=self.lift_motor.ControlMode.MotionMagic, value=self.set_pos)
+        self.set_pos = self.BOTTOM_HEIGHT
 
     def move(self):
```

Comment:
Why is there two functions to do this, rather than just one which accepts the height as an argument and sets the `set_pos` variable to what it needs to be? What is the purpose of `default_height`?